l3kernel/l3token.dtx

```diff
@@ -2914,6 +2859,57 @@ end
 %    \end{macrocode}
 % \end{macro}
 %
+% \begin{macro}{\peek_remove_filler:n}
+% \begin{macro}{\@@_remove_filler:w}
+% \begin{macro}{\@@_remove_filler:}
+% \begin{macro}{\@@_remove_filler_expand:w}
+%   Here we expand the input, removing spaces and \cs{scan_stop:} tokens until
+%   we reach a non-expandable token. At that stage we re-insert the payload.
+%   To deal with the problem of |&| tokens, we have to put the align-safe
+%   group in the correct place.
+%    \begin{macrocode}
+\cs_new_protected:Npn \peek_remove_filler:n #1
+  {
+    \cs_set:Npn \@@_true_aux:w { \@@_remove_filler:w }
+    \cs_set:Npx \@@_false:w
+      {
+        \exp_not:N \group_align_safe_end:
+        \exp_not:n {#1}
+      }
+    \group_align_safe_begin:
+    \@@_remove_filler:w
+  }
+\cs_new_protected:Npn \@@_remove_filler:w
+  {
+    \exp_after:wN \peek_after:Nw \exp_after:wN \@@_remove_filler:
+    \exp:w \exp_end_continue_f:w
+  }
+\cs_new_protected:Npn \@@_remove_filler:
+  {
+    \if_catcode:w \exp_not:N \l_peek_token \c_space_token
+      \exp_after:wN \@@_true_remove:w
+    \else:
+      \if_meaning:w \l_peek_token \scan_stop:
+        \exp_after:wN \exp_after:wN \exp_after:wN
+          \@@_true_remove:w
+      \else:
+        \exp_after:wN \exp_after:wN \exp_after:wN
+          \@@_remove_filler_expand:w
+      \fi:
+    \fi:
+  }
+\cs_new_protected:Npn \@@_remove_filler_expand:w
+  {
+    \token_if_expandable:NTF \l_peek_token
+      { \@@_remove_filler:w }
+      { \@@_false:w }
+  }
```

Comment:
`\token_if_expandable:NTF` (wrongly in my opinion) rejects undefined control sequences as being non-expandable.  This means the `remove_filler` code will stop when seeing an undefined control sequence.  Please document this edge case.

Reply:
@blefloch why do you think that it is wrong to stop scanning if you hit an undefined csname? It isn't a filler part that would be skipped over in the TeX sense and I can't think of a use case where you would want to skip over undefined csnames. Or am I miss something?

Reply:
I've documented but I don't see how one could describe an undefined control sequence as expandable: `\expandafter\foo\undefined` is an error.

Reply:
I think the point here is that while scanning for a filler, TeX will expand the undefined cs, throw an (expandable) error, and proceed, so `\expanded \foo {a}` expands to just `a` plus the error.  I'd expect `\peek_remove_filler:n {} \foo a` to do the same.

Reply:
I've adjusted the code so we get the TeX error.